syntax = "proto3";

import "Base.proto";
import "Task.proto";
import "Input.proto";
import "Statistic.proto";
import "Dependency.proto";

package dra;

// The DependencyRPC service definition.
service DependencyRPC {

    // DRA and syz-manager
    rpc GetVMOffsets (Empty) returns (Empty) {
    }
    rpc SendBasicBlockNumber (Empty) returns (Empty) {
    }
    rpc GetNewInput (Empty) returns (Inputs) {
    }
    rpc SendDependency (Dependency) returns (Empty) {
    }
    rpc GetCondition (Empty) returns (Conditions) {
    }
    rpc SendWriteAddress (WriteAddresses) returns (Empty) {
    }

    //syz-fuzzer and syz-manager
    rpc Connect (Empty) returns (Empty) {
    }
    rpc SendNewInput (Input) returns (Empty) {
    }
    rpc GetTasks (Empty) returns (Tasks) {
    }
    rpc GetBootTasks (Empty) returns (Tasks) {
    }
    rpc ReturnTasks (Tasks) returns (Empty) {
    }
    rpc SendBootInput (Input) returns (Empty) {
    }
    rpc SendUnstableInput (UnstableInput) returns (Empty) {
    }
    rpc SendLog (Empty) returns (Empty) {
    }
    rpc sendStat (Statistic) returns (Empty) {
    }
    rpc GetNeed (Empty) returns (Empty) {
    }
    rpc SendNeedInput (Input) returns (Empty) {
    }
}

message Corpus {
    map<string, Input> input = 1;
    map<uint32, UncoveredAddress> uncovered_address = 4;
    map<uint32, UncoveredAddress> covered_address = 2;
    map<uint32, WriteAddress> write_address = 5;
    map<string, FileOperations> file_operations = 6;
    Tasks tasks = 7;
    Tasks high_task = 8;
    Tasks boot_task = 9;
    map<string, Input> new_input = 11;
}